### ISSUE TITLE
PATCH methods not supported by JS but in templates

### ISSUE THREAD

**Issue by tarjei** [FriendsOfSymfony/FOSCommentBundle#354]
TL;DR comments.js uses POST on methods that require PATCH.

There are two methods in the API that require use of the PATCH HTTP method:
    fos_comment_patch_thread_comment_state           PATCH       ANY    ANY  /comments/api/threads/{id}/comments/{commentId}/state.{_format}
    fos_comment_patch_thread_commentable             PATCH       ANY    ANY  /comments/api/threads/{id}/commentable.{_format}

At least one of these(fos_comment_patch_thread_comment_state) is referenced and used by comments.js. 

Another problem with this is that the fos_comment_patch_thread_comment_state returns a 302 at the end of the request, causing JQuery to do a PATCH request against a resource that doesn't support this operation.

I can create a PR for this, but I want to get some feedback to what is the best solution first:

a) Changing the JS to support PATCH + changing the responses to not redirect.
b) Changing the PATCH methods to use/support POST instead.

**Comment by tarjei** [comment]
Note, this was me missing Request::enableHttpMethodParameterOverride(); in `web/app.php`.
